File: backend/onGo-domain/src/main/kotlin/com/ongo/domain/subtitletranslation/SubtitleTranslation.kt
package com.ongo.domain.subtitletranslation

import java.time.LocalDateTime

data class SubtitleTranslation(
    val id: Long = 0,
    val workspaceId: Long,
    val videoTitle: String,
    val sourceLanguage: String,
    val targetLanguage: String,
    val status: String = "PENDING",
    val progress: Int = 0,
    val sourceLineCount: Int = 0,
    val translatedLineCount: Int = 0,
    val quality: Int = 0,
    val cost: Int = 0,
    val createdAt: LocalDateTime = LocalDateTime.now(),
    val completedAt: LocalDateTime? = null,
    val updatedAt: LocalDateTime = LocalDateTime.now(),
)

File: frontend/src/types/subtitleTranslation.ts
export interface SubtitleTranslation {
  id: number
  videoTitle: string
  sourceLanguage: string
  targetLanguage: string
  status: 'PENDING' | 'TRANSLATING' | 'COMPLETED' | 'FAILED' | 'REVIEWING'
  progress: number
  sourceLineCount: number
  translatedLineCount: number
  quality: number
  cost: number
  createdAt: string
  completedAt?: string
}

export interface TranslationLine {
  id: number
  translationId: number
  lineNumber: number
  startTime: string
  endTime: string
  sourceText: string
  translatedText: string
  isEdited: boolean
}

export interface SupportedLanguage {
  code: string
  name: string
  nativeName: string
}

export interface SubtitleTranslationSummary {
  totalTranslations: number
  completedCount: number
  inProgressCount: number
  totalCreditsUsed: number
  avgQuality: number
}

export interface CreateTranslationRequest {
  videoId: number
  videoTitle: string
  sourceLanguage: string
  targetLanguage: string
}
